logbus/logstreamer/logstreamer.go

```diff
@@ -124,40 +218,66 @@ func (ls *LogStreamer) tryStream(ctx context.Context) (bool, error) {
 	return false, nil
 }
 
-// Write writes the given delta to the log streamer.
+// WriteAsync queues delta up for writing and returns immediately. The returned
+// channel will be closed when delta has been passed off to the buffer.
+//
+// This is different than Write in two important ways:
+//
+// 1. It will never block.
+//
+// 2. If the buffer is full, it cannot guarantee the order that delta will
```

Comment:
do we have a use case for WriteAsync?

The out of order writes here seem to go against what we'll be using this for.

I do see that we have a regular `Write`, which doesn't have this limitation, but i'm worried someone might use `WriteAsync` one day without reading this comment.

Reply:
Yeah, I thought about that. The main thing is that I could not test it without `WriteAsync`, and I hate putting tests in the main (non-`_test`) package. I was wondering what others' thoughts were about that.

Quoting the text above (in the description of this PR):
> Technically, we could unexport the WriteAsync method, but that would require the logstreamer_chaos_test.go file to be in the logstreamer package (instead of the logstreamer_test package), and that always makes me feel a little weird. Providing a test case that users of the package cannot exercise (because they don't have access to the same methods) feels like a hack.